warc_metadata_sidecar.py

```diff
@@ -0,0 +1,261 @@
+                if text_format_mimes.search(fido.mime):
+                    payload.seek(0)
+                    bytes_payload = payload.read()
+                    result_dict = find_character_set(bytes_payload)
+                    lang_cld = find_language(bytes_payload)
+                    text_mime += 1
+                    if not lang_cld:
+                        string_payload = '{0} {1}\n{2} {3}\n{4} {5}'.format(
+                            mime_title, fido.mime,
+                            puid_title, fido.puid,
+                            charset_title, result_dict
+                            )
+                    else:
+                        string_payload = '{0} {1}\n{2} {3}\n{4} {5}\n{6} {7}'.format(
+                            mime_title, fido.mime,
+                            puid_title, fido.puid,
+                            charset_title, result_dict,
+                            language_title, lang_cld
+                            )
+                else:
+                    non_text += 1
+                    string_payload = '{0} {1}\n{2} {3}'.format(
+                        mime_title, fido.mime,
+                        puid_title, fido.puid
+                        )
+            else:
+                non_mime += 1
+                string_payload = ''
+
+            writer = WARCWriter(output, gzip=False)  # TODO: gzip will equal True
+            meta_record = writer.create_warc_record(url,
+                                                    'metadata',
+                                                    payload=io.BytesIO(string_payload.encode()),
+                                                    warc_headers_dict=warc_dict
+                                                    )
+            writer.write_record(meta_record)
+        # Delete sidecar file if the only record is 'warcinfo'.
+        if record_count == 1:
+            if rec_type:
+                os.remove(warc_file_path)
+                logging.info('Deleted sidecar, no records to collect.')
+        else:
+            logging.info('Finished creating sidecar in %s',
+                         str(timedelta(seconds=(time.time() - start))))
+            logging.info('Found %s record(s)', record_count)
+    print('Mimes: ' + str(text_mime + non_text) + ' Non Mimes: ' + str(non_mime))
+    print('Total Records for this WARC file: ', total_records)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'warc_file',
+        action='store',
+        help='a WARC file that will be used to generate a sidecar with metadata'
+    )
+    parser.add_argument(
+        'archive_dir',
+        action='store',
+        help='a directory where the sidecar and log will be stored',
+    )
+    parser.add_argument(
+        '--operator',
+        action='store',
+        default=None,
+        help='a name or name and email address of the person who created this WARC file'
```

Comment:
Perhaps 'a name or name and email address of the person running warc-metadata-sidecar'.